lib/vault/api/auth.rb

```diff
@@ -13,6 +13,10 @@ def auth
   end
 
   class Authenticate < Request
+
+    # canary header used for aws_ec2_iam
+    IAM_SERVER_ID_HEADER = "canaryHeaderValue".freeze
```

Comment:
I'm not a fan of this. Right now, it happens to work in Vault that if you haven't configured the server ID header value, it will accept any or no header value. This isn't great behavior, and might be something that I'd want to change in the future, so I'd rather not have the ruby client depend on it. Can you just make the default value `nil` and then just add the header value into the `vault_headers` if it's non-`nil`.